### PR TITLE
chore: update the roadmap

### PR DESCRIPTION
**Pull request by archerny**
# Which issue does this PR close?

None.

# Rationale for this change
 
The roadmap is too old. Update our recent work about CeresDB 1.0 and subsequent working schedule.

# What changes are included in this PR?

The doc about roadmap.

# Are there any user-facing changes?

No.

# How does this change test

No need.

### CODE REVIEW

File: docs/guides/src/dev/roadmap.md [hunk 1]
```diff
@@ -14,18 +14,28 @@
...
+- [x] Improve the query performance for traditional time series workloads.
+- [x] Support dynamic migration of tables in cluster mode.
 
-### Afterward
-- [ ] Support Prometheus protocol.
-- [ ] Implement UDF framework which makes CeresDB more extensible.
+### v1.0.0
+- [ ] Formally release CeresDB and its SDKs. All breaking changes will be finished.
+- [ ] Finish the majority of work related to PartitionTable.
+- [ ] Various efforts to improve query performance, especially for cloud-native cluster mode. Theses works includes:
```
Comment (ZuLiangWang) [inline docs/guides/src/dev/roadmap.md:30]:
```suggestion
- [ ] Various efforts to improve query performance, especially for cloud-native cluster mode. These works includes:
```

File: docs/guides/src/dev/roadmap.md [hunk 2]
```diff
@@ -14,18 +14,28 @@
...
+- [x] Release Golang client.
+- [x] Improve the query performance for traditional time series workloads.
+- [x] Support dynamic migration of tables in cluster mode.
 
-### Afterward
-- [ ] Support Prometheus protocol.
-- [ ] Implement UDF framework which makes CeresDB more extensible.
+### v1.0.0
+- [ ] Formally release CeresDB and its SDKs. All breaking changes will be finished.
+- [ ] Finish the majority of work related to PartitionTable.
```
Comment (ZuLiangWang) [inline docs/guides/src/dev/roadmap.md:29]:
```suggestion
- [ ] Finish the majority of work related to Table Partitioning.
```

File: docs/guides/src/dev/roadmap.md [hunk 3]
```diff
@@ -14,18 +14,28 @@
...
+- [ ] Various efforts to improve query performance, especially for cloud-native cluster mode. Theses works includes:
+    - Multi-tier cacheing.
+    - Introduce various methods to reduce the data fetched from remote storage (improve the accuracy of SST data filtering).
+    - Increase the parallelism while fetching data from remote object-store.
+    - Add some resource-controlling methods.
```
Comment (ShiKaiWi) [inline docs/guides/src/dev/roadmap.md:34]: It seems this does nothing with the query performance improvement.

File: docs/guides/src/dev/roadmap.md [hunk 4]
```diff
@@ -14,18 +14,28 @@
...
+- [x] Support dynamic migration of tables in cluster mode.
 
-### Afterward
-- [ ] Support Prometheus protocol.
-- [ ] Implement UDF framework which makes CeresDB more extensible.
+### v1.0.0
+- [ ] Formally release CeresDB and its SDKs. All breaking changes will be finished.
+- [ ] Finish the majority of work related to PartitionTable.
+- [ ] Various efforts to improve query performance, especially for cloud-native cluster mode. Theses works includes:
+    - Multi-tier cacheing.
```
Comment (ShiKaiWi) [inline docs/guides/src/dev/roadmap.md:31]:
```suggestion
    - Multi-tier cache.
```
Or fix the typo:
```suggestion
    - Multi-tier caching.
```

File: docs/guides/src/dev/roadmap.md [hunk 5]
```diff
@@ -14,18 +14,30 @@
...
-- [ ] Support more underlying storages for adaptivity. For example, the underlying storage could be a distributed file system.
-- [ ] Various tools that ease the use of CeresDB. For example, data import and export tools.
+### v1.0.0
+- [ ] Formally release CeresDB and its SDKs. All breaking changes will be finished.
+- [ ] Finish the majority of work related to `Table Partitioning`.
+- [ ] Various efforts to improve query performance, especially for cloud-native cluster mode. These works includes:
+    - Multi-tier cache.
+    - Introduce various methods to reduce the data fetched from remote storage (improve the accuracy of SST data filtering).
+    - Increase the parallelism while fetching data from remote object-store.
+- [ ] Improve data ingestion performance with some resource-controlling methods.
```
Comment (ShiKaiWi) [inline docs/guides/src/dev/roadmap.md:34]:
```suggestion
- [ ] Improve data ingestion performance by introducing resource control over compaction.
```

File: docs/guides/src/dev/roadmap.md [hunk 6]
```diff
@@ -14,18 +14,30 @@
...
+- [x] Improve the query performance for traditional time series workloads.
+- [x] Support dynamic migration of tables in cluster mode.
 
-### Afterward
-- [ ] Support Prometheus protocol.
-- [ ] Implement UDF framework which makes CeresDB more extensible.
-- [ ] Support more underlying storages for adaptivity. For example, the underlying storage could be a distributed file system.
-- [ ] Various tools that ease the use of CeresDB. For example, data import and export tools.
+### v1.0.0
+- [ ] Formally release CeresDB and its SDKs. All breaking changes will be finished.
```
Comment (ShiKaiWi) [inline docs/guides/src/dev/roadmap.md:28]:
```suggestion
- [ ] Formally release CeresDB and its SDKs with all breaking changes finished.
```

File: docs/guides/src/dev/roadmap.md [hunk 7]
```diff
@@ -14,18 +14,30 @@
...
+- [ ] Finish the majority of work related to `Table Partitioning`.
+- [ ] Various efforts to improve query performance, especially for cloud-native cluster mode. These works includes:
+    - Multi-tier cache.
+    - Introduce various methods to reduce the data fetched from remote storage (improve the accuracy of SST data filtering).
+    - Increase the parallelism while fetching data from remote object-store.
+- [ ] Improve data ingestion performance with some resource-controlling methods.
+
+### Afterwards
+With an in-depth understanding of the time-series database and its various use cases, the majority of our work will focus on performance, reliability, scalability, ease of use, and collaborations with open-source communities.
+- [ ] Add utilities that support promQL, influxQL, opentsdb protocol, and so on.
```
Comment (ZuLiangWang) [inline docs/guides/src/dev/roadmap.md:38]:
```suggestion
- [ ] Add utilities that support `PromQL`, `InfluxQL`, `OpenTSDB` protocol, and so on.
```

File: docs/guides/src/dev/roadmap.md [hunk 8]
```diff
@@ -14,18 +14,30 @@
...
+- [ ] Provide basic utilities for operation and maintenance. Specifically, include the following:
```
Comment (ShiKaiWi) [inline docs/guides/src/dev/roadmap.md:39]:
```suggestion
- [ ] Provide basic utilities for operation and maintenance. Specifically, the following are included:
```

File: docs/guides/src/dev/roadmap.md [hunk 9]
```diff
@@ -14,18 +14,30 @@
...
+    - Deployment tools that fit well for both Kubernetes and bare servers.
```
Comment (ZuLiangWang) [inline docs/guides/src/dev/roadmap.md:40]:
```suggestion
    - Deployment tools that fit well for both `Kubernetes` and `Bare` servers.
```

Comment (archerny) [reply docs/guides/src/dev/roadmap.md]: The word bare is not terminology.

**Review by ShiKaiWi — APPROVED** [review]
LGTM 👍

**Pull request merged**